google/cloud/bigquery/v2/minimal/internal/job_stats.cc

```diff
@@ -0,0 +1,134 @@
+
+namespace google {
+namespace cloud {
+namespace bigquery_v2_minimal_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+EvaluationKind EvaluationKind::UnSpecified() {
+  return EvaluationKind{"EVALUATION_KIND_UNSPECIFIED"};
+}
+EvaluationKind EvaluationKind::Statement() {
+  return EvaluationKind{"STATEMENT"};
+}
+EvaluationKind EvaluationKind::Expression() {
+  return EvaluationKind{"EXPRESSION"};
+}
+
+void to_json(nlohmann::json& j, JobStatistics const& s) {
+  j = nlohmann::json{
+      {"total_bytes_processed", s.total_bytes_processed},
+      {"num_child_jobs", s.num_child_jobs},
+      {"total_modified_partitions", s.total_modified_partitions},
+      {"parent_job_id", s.parent_job_id},
+      {"session_id", s.session_id},
+      {"transaction_id", s.transaction_id},
+      {"reservation_id", s.reservation_id},
+      {"row_level_security_applied", s.row_level_security_applied},
+      {"data_masking_applied", s.data_masking_applied},
+      {"start_time",
+       std::chrono::duration_cast<std::chrono::milliseconds>(s.start_time)
+           .count()},
+      {"end_time",
+       std::chrono::duration_cast<std::chrono::milliseconds>(s.end_time)
+           .count()},
+      {"creation_time",
+       std::chrono::duration_cast<std::chrono::milliseconds>(s.creation_time)
+           .count()},
+      {"total_slot_time",
+       std::chrono::duration_cast<std::chrono::milliseconds>(s.total_slot_time)
+           .count()},
+      {"final_execution_duration",
+       std::chrono::duration_cast<std::chrono::milliseconds>(
+           s.final_execution_duration)
+           .count()},
+      {"completion_ratio", s.completion_ratio},
+      {"quota_deferments", s.quota_deferments},
+      {"script_statistics", s.script_statistics},
+      {"job_query_stats", s.job_query_stats}};
+}
+
+void from_json(nlohmann::json const& j, JobStatistics& s) {
+  if (j.contains("total_bytes_processed"))
+    j.at("total_bytes_processed").get_to(s.total_bytes_processed);
+  if (j.contains("num_child_jobs"))
+    j.at("num_child_jobs").get_to(s.num_child_jobs);
+  if (j.contains("total_modified_partitions"))
+    j.at("total_modified_partitions").get_to(s.total_modified_partitions);
+  if (j.contains("parent_job_id"))
+    j.at("parent_job_id").get_to(s.parent_job_id);
+  if (j.contains("session_id")) j.at("session_id").get_to(s.session_id);
+  if (j.contains("transaction_id"))
+    j.at("transaction_id").get_to(s.transaction_id);
+  if (j.contains("reservation_id"))
+    j.at("reservation_id").get_to(s.reservation_id);
+  if (j.contains("row_level_security_applied"))
+    j.at("row_level_security_applied").get_to(s.row_level_security_applied);
+  if (j.contains("data_masking_applied"))
+    j.at("data_masking_applied").get_to(s.data_masking_applied);
+  if (j.contains("completion_ratio"))
+    j.at("completion_ratio").get_to(s.completion_ratio);
+  if (j.contains("start_time")) {
+    std::int64_t millis;
```

Comment:
I think you should consider a clean up PR where you refactor these conversions to and from milliseconds to something more like:

```cc
  s.start_time = FromMillisecondField(j, "start_time");
```

or even

```cc
  FromMillisecondField(s.start_time, j, "start_time");
```

where the helper would be something like:

```cc 
void FromMillisecondField(std::chrono::milliseconds& field, nlohman::json const& j, char const* name) {
  auto const l = j.find(name);
  if (l == j.end()) return;
  std::int64_t m;
  l->get_to(m);
  field = std::chrono::milliseconds(m);
}
```